wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt

```diff
@@ -0,0 +1,26 @@
+package com.squareup.wire.schema
+
+import com.squareup.wire.schema.ProtoFile.Syntax
+import com.squareup.wire.schema.ProtoFile.Syntax.PROTO_2
+import com.squareup.wire.schema.ProtoFile.Syntax.PROTO_3
+
+/** A set of rules which defines schema requirements for a specific [Syntax]. */
+interface SyntaxRules {
+  fun enumRequiresZeroValueAtFirstPosition(): Boolean = false
+
+  companion object {
+    fun get(syntax: Syntax?): SyntaxRules {
+      return when (syntax) {
+        PROTO_2 -> PROTO_2_SYNTAX_RULES
+        PROTO_3 -> PROTO_3_SYNTAX_RULES
+        null -> PROTO_2_SYNTAX_RULES
+      }
+    }
+
+    internal val PROTO_2_SYNTAX_RULES = object : SyntaxRules {}
```

Comment:
I wonder if we should avoid default values and always set it explicitly.

Reply:
When designing systems like this I prefer obviousness over terseness. Default values don’t save much code but do introduce indirection so I’d avoid ’em

Reply:
Ok! Fixed 'em.